pipeline/eBOSS_QSO/QuasarMapper.py

```diff
@@ -0,0 +1,191 @@
+
+import os
+import pandas as pd
+import collections
+import numpy as np
+import healpy as hp
+
+import pymaster as nmt
+
+
+class QuasarMapper:
+    def __init__(self): 
+        #Inputs: the four fits files from eBOSS DR14
+        
+        data_path = '/home/zcapjru/PhD/Data/'
+        #Check for data files:
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_data-NGC-vDR16.fits') :
+            print('NGC data found')
+            self.NGC_data =  fits.open(data_path + 'eBOSS_QSO_clustering_data-NGC-vDR16.fits')
+        else :
+            print('missing NGC data')
+
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_data-SGC-vDR16.fits') :
+            print('SGC data found')
+            self.SGC_data =  fits.open(data_path + 'eBOSS_QSO_clustering_data-SGC-vDR16.fits')
+        else :
+            print('missing SGC data')
+           
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_random-NGC-vDR16.fits') :
+            print('NGC random data found')
+            self.NGC_r_data =  fits.open(data_path + 'eBOSS_QSO_clustering_random-NGC-vDR16.fits')
+        else :
+            print('missing NGC random data')
+           
+        if os.path.exists(data_path+'eBOSS_QSO_clustering_random-SGC-vDR16.fits') :
+            print('SGC random data found')
+            self.SGC_r_data =  fits.open(data_path + 'eBOSS_QSO_clustering_random-SGC-vDR16.fits')
+        else :
+            print('missing SGC random data')
+        
+
+        self.nside = 512
+        #set resolution
+        
+        self.bands = nmt.NmtBin.from_nside_linear(self.nside, 20)
+        #set l's per band
+        
+        self.NGC = Table.read(self.NGC_data).to_pandas()
+        self.NGC_r = Table.read(self.NGC_r_data).to_pandas()
+        self.SGC = Table.read(self.SGC_data).to_pandas()
+        self.SGC_r = Table.read(self.SGC_r_data).to_pandas()
+        #from .fits to table to managable pandas frames
+
+        #Merge NGC + SGC
+        self.edges = [1.5] #inner bins edges
+        self.whole =  pd.concat([self.NGC, self.SGC])
+        self.whole_r =  pd.concat([self.NGC_r, self.SGC_r])
+        
+        self.binned   = self.bin_z(self.whole ,  self.edges)
+        self.binned_r = self.bin_z(self.whole_r ,  self.edges)
+        #arrays containing the binned data for each field
+        
+        self.maps = {}
+        for i in range(len(self.edges)+1):
+            self.nmean_field  =  self.get_nmean_map(self.binned[i])
+            self.nmean_random =  self.get_nmean_map(self.binned_r[i])
+            self.alpha = self.get_alpha(self.nmean_field, self.nmean_random)
+            
+            self.maps["bin_{}".format(i)+"_delta"]= self.get_delta_map(self.nmean_field, self.nmean_random, self.alpha) 
+            self.maps["bin_{}".format(i)+"_nmean"]= self.nmean_random*self.alpha
+            self.maps["bin_{}".format(i)+"_mask"] = self.nmean_random
+               
+        print('Maps computed succesfully')

```

Comment:
No printing (or, if you do, enclose in something like `if self.verbose`, or something like that).